src/misc/unicode.cpp

```diff
@@ -451,712 +653,878 @@ static bool wide_to_code_page(const std::vector<uint16_t> &str_in,
+				return;
+			}
+			if (!is_combining_mark(code_point_2)) {
+				error_not_combining_mark(idx + 1,
+				                         file_name_decomposition,
+				                         line_num);
+				return;
+			}
+			new_rules[code_point_1].AddMark(code_point_2);
+		}
+	}
+
+	if (!check_import_status(in_file, file_name_decomposition, new_rules.empty())) {
+		return;
+	}
+
+	// Reading/parsing succeeded - use the rules
+	decomposition_rules = new_rules;
 }
 
-static void import_mapping_ascii(const std_fs::path &path_root)
+static void import_mapping_ascii(const std_fs::path& path_root)
 {
-    // Import fallback mapping, from UTH-8 to 7-bit ASCII;
-    // this mapping will only be used if everything else fails
+	// Import fallback mapping, from Unicode to 7-bit ASCII;
+	// this mapping will only be used if everything else fails
 
-    // Open the file
-    auto in_file = open_mapping_file(path_root, file_name_ascii);
-    if (!in_file)
-        return;
+	// Open the file
+	auto in_file = open_mapping_file(path_root, file_name_ascii);
+	if (!in_file) {
+		return;
+	}
 
-    // Read and parse
-    std::string line_str = "";
-    size_t line_num      = 0;
+	// Read and parse
+	std::string line_str = "";
+	size_t line_num      = 0;
 
-    code_page_mapping_t new_mapping_ascii;
+	code_page_mapping_t new_mapping_ascii;
 
-    while (get_line(in_file, line_str, line_num)) {
-        std::vector<std::string> tokens;
-        if (!get_tokens(line_str, tokens))
-            continue; // empty line
+	while (get_line(in_file, line_str, line_num)) {
+		std::vector<std::string> tokens;
+		if (!get_tokens(line_str, tokens)) {
+			continue; // empty line
+		}
 
-        uint16_t code_point = 0;
-        uint8_t character   = 0;
+		uint16_t code_point = 0;
+		uint8_t character   = 0;
 
-        if (tokens.size() != 2 ||
-              !get_hex_16bit(tokens[0], code_point) ||
-            !get_ascii(tokens[1], character)) {
-            error_parsing(file_name_ascii, line_num);
-            return;
-        }
+		if (tokens.size() != 2 || !get_hex_16bit(tokens[0], code_point) ||
+		    !get_ascii(tokens[1], character)) {
+			error_parsing(file_name_ascii, line_num);
+			return;
+		}
 
-        new_mapping_ascii[code_point] = character;
-    }
+		new_mapping_ascii[code_point] = character;
+	}
 
-    if (!check_import_status(in_file, file_name_ascii, new_mapping_ascii.empty()))
-         return;
+	if (!check_import_status(in_file, file_name_ascii, new_mapping_ascii.empty())) {
+		return;
+	}
 
-    // Reading/parsing succeeded - use the mapping
-    mapping_ascii = new_mapping_ascii;
+	// Reading/parsing succeeded - use the mapping
+	mapping_ascii = new_mapping_ascii;
 }
 
 static uint16_t deduplicate_code_page(const uint16_t code_page)
 {
-    const auto it = config_duplicates.find(code_page);
+	const auto it = config_duplicates.find(code_page);
+
+	if (it == config_duplicates.end()) {
+		return code_page;
+	} else {
+		return it->second;
+	}
```

Comment:
The `else if` can become a stand-alone `if` because the prior `if` returns.

Reply:
Done.